### ISSUE TITLE
Set up JavaScript and Typescript linting and formatting tools

### ISSUE THREAD

**Issue by Dripcoding** [SasanLabs/VulnerableApp-facade#64]
**Is your feature request related to a problem? Please describe.**
I would like to automate linting and formatting when contributing code. This will make collaboration easier and helps ensure code-quality.

**Describe the solution you'd like**
I would like ESLint, TSLint, Husky, Prettier to be configured for the project locally and as part of the ci/cd pipeline.

**Describe alternatives you've considered**
I've also considered looking in to using code quality tools available on the Github Actions marketplace.

**Additional context**
NA

**Would you like to help fix this issue?**
Yes, I'd like to work on this feature.

**Comment by preetkaran20** [comment]
We already have prettier configured https://github.com/SasanLabs/VulnerableApp-facade/blob/main/facade-app/package.json#L27

If you still think we can make it better or integrate with npm build please go ahead

**Comment by kkreine** [comment]
Hi @preetkaran20, I'd be happy to work on this. I think there is value in having both ESLint and Prettier, but let me know what you think best suits the need of the project. Given that their rules can conflict with each other, I'll spend some time reviewing their configs and defaults to minimize that. I would be happy to set up a husky hook to run ESLint and/or Prettier on commit too, so that devs don't have to run the extra prettier command before committing.

**Comment by preetkaran20** [comment]
@kkreine please go ahead with the task. 

cc: @Dripcoding

**Comment by kkreine** [comment]
I'll be working on this this weekend.

**Comment by kkreine** [comment]
Hi @preetkaran20, @Dripcoding, sorry for being a little late on this, but I have Husky and ESLint configured and working with Prettier. However, there are a few default ESLint rules the code is breaking, including:
- Don't use `{}` as a type. `{}` actually means "any non-nullish value" (`@typescript-eslint/ban-types`)
- Unexpected any. Specify a different type (`@typescript-eslint/no-explicit-any`)
- Missing "key" prop for element in iterator (`react/jsx-key`)
- and some unused vars (`@typescript-eslint/no-unused-vars`)

I can send over more detail & files affected if you'd like.

Would you like me to address these issues? I'd be happy to, and many are pretty straightforward, but I wanted to check before I go beyond the initial scope of this issue.

**Comment by preetkaran20** [comment]
@kkreine thanks for pointing those issues, please go ahead with fixing those.

thanks,
Karan

**Comment by preetkaran20** [comment]
@kkreine are you still working on this ?

**Comment by rarestpreet** [comment]
@preetkaran20 does the pipeline ensure consistent code format and styling using any tools?

**Comment by preetkaran20** [comment]
I think no. Can you please validate if possible?

**Comment by rarestpreet** [comment]
@preetkaran20, you're right. The pipeline only build the TS/JS module and test them but does not enforce usage of consistent code format/style.
